File: javascript/2533.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.split('\n');
var prompt = function(texto) { return lines.shift();};

while (true) {
  var cases = parseInt(prompt());
  if (isNaN(cases)) {
    break;
  }
  var sumT = 0;
  var sumW = 0;

  for (let i = 0; i < cases; i++) {
    var [grade, workload] = prompt().split(" ").map(Number);
    sumT = sumT + grade * workload;
    sumW = sumW + workload;
  }

  var API = sumT / (sumW * 100);
  console.log(API.toFixed(4));
}

File: java/2533.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        
        while (sc.hasNext()) {
            String firstLine = sc.nextLine().trim();
            if (firstLine.isEmpty()) continue;
            
            int cases;
            try {
                cases = Integer.parseInt(firstLine);
            } catch (NumberFormatException e) {
                break;
            }
            
            double sumT = 0.0;
            int sumW = 0;
            
            for (int i = 0; i < cases; i++) {
                int grade = sc.nextInt();
                int workload = sc.nextInt();
                sc.nextLine(); // consumir a quebra de linha
                
                sumT += grade * workload;
                sumW += workload;
            }
            
            double API = sumT / (sumW * 100.0);
            System.out.printf("%.4f%n", API);
        }
        
        sc.close();
    }
}
